sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -531,44 +531,52 @@ pi_result _pi_device::initialize() {
           (ZeDevice, &numQueueGroups, QueueProperties.data()));
 
   int ComputeGroupIndex = -1;
-  for (uint32_t i = 0; i < numQueueGroups; i++) {
-    if (QueueProperties[i].flags &
-        ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE) {
-      ComputeGroupIndex = i;
-      break;
-    }
-  }
-  // How is it possible that there are no "compute" capabilities?
-  if (ComputeGroupIndex < 0) {
-    return PI_ERROR_UNKNOWN;
-  }
-  ZeComputeQueueGroupIndex = ComputeGroupIndex;
-  ZeComputeQueueGroupProperties = QueueProperties[ComputeGroupIndex];
 
-  int CopyGroupIndex = -1;
-  const char *CopyEngine = std::getenv("SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE");
-  bool UseCopyEngine = (!CopyEngine || (std::stoi(CopyEngine) != 0));
-  if (UseCopyEngine) {
+  // initialize a sub-sub-devices with it's own Ordinal
+  if (SubSubDeviceOrdinal >= 0) {
+    ComputeGroupIndex = SubSubDeviceOrdinal;
+  } else {
     for (uint32_t i = 0; i < numQueueGroups; i++) {
-      if (((QueueProperties[i].flags &
-            ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE) == 0) &&
-          (QueueProperties[i].flags &
-           ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COPY)) {
-        CopyGroupIndex = i;
+      if (QueueProperties[i].flags &
+          ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE) {
+        ComputeGroupIndex = i;
```

Comment:
We essentially pick the first compute group, say 0, and send all compute there.
How sub-device (with such chosen 0 index) is different from a sub-sub-device (with 0 ordinal assigned)?
They seem to be being exactly the same thing.

Reply:
@smaslov-intel  correct, it's the same thing. In both cases:

subdevice 0's compute queue group 0's index 0 = sub-sub-device with the compute engine 0 from queue group 0.

so all that you are doing in this patch is to create a sub-sub-device out of each index. So instead of having:

1 subdevice with N engines

you would have

N sub-sub-devices with 1 engine each

Reply:
> 1 subdevice with N engines

It appears that we don't have this today either. Our current sub-device is a sub-sub-device with 1 engine (first compute one). Is there a way to have a sub-device that is implicitly scaling to all underlying engines?